### PR TITLE
Update private primitive `colors`

### LINKED ISSUE #10414: Add private primitive `color` token layer

### PR DESCRIPTION
**Pull request by aaronccasanova**
### WHY are these changes introduced?

Fixes #10414

### WHAT is this pull request doing?

- Removed transparency from the `gray` color scale
- Added a new private primitive `blackAlpha` color scale
- Updated typings to better enforce our conventions for opaque and transparent scales
- Find/replaced opaque `gray` color scale references with the original computed member access syntax
- Updated transparent `gray` color scale references to use the new `blackAlpha` color scale

### CODE REVIEW

File: polaris-tokens/src/token-groups/color.ts [hunk 1]
```diff
@@ -767,61 +767,61 @@ export const color: {
...
     description: '',
   },
   'color-bg-transparent-hover-experimental': {
-    value: colorsExperimental.gray[16]('0.05'),
+    value: colorsExperimental.blackAlpha[5],
     description: '',
   },
   'color-bg-transparent-active-experimental': {
-    value: colorsExperimental.gray[16]('0.07'),
+    value: colorsExperimental.blackAlpha[7],
```
Comment (sarahill) [inline polaris-tokens/src/token-groups/color.ts]:
```suggestion
    value: colorsExperimental.blackAlpha[6],
```

Looks a smidge dark with `7`

**Review by sarahill — APPROVED** [review]
Overall looks good just made one suggestion ✅

**Review by lgriffee — APPROVED** [review]

**Pull request merged**